unmarshal_test.go

```diff
@@ -661,9 +661,129 @@ name = "a"
 			}{
 				Name: "a",
 			},
-			fail: `error applying default value "a" to list: error converting default value "a" to int`,
+			fail: `error parsing default value: error applying "a" to list: error parsing default value: error converting "a" to int`,
 		}}
 
 	runTests(t, tests)
+}
+
+func TestEnumValidCases(t *testing.T) {
+	tests := []test{
+		{
+			name: "enum valid cases",
+			hcl: `
+name = "test"
+str_val = "a"
+int_val = 5
+float_val = 2.11
+`,
+			dest: struct {
+				Name     string  `hcl:"name"`
+				StrVal   string  `hcl:"str_val" enum:"a,b,c"`
+				IntVal   int64   `hcl:"int_val" enum:"2,5,8"`
+				FloatVal float64 `hcl:"float_val" enum:"2.11,5.32,8.91"`
+			}{
+				Name:     "test",
+				StrVal:   "a",
+				IntVal:   5,
+				FloatVal: 2.11,
+			},
+		},
+	}
+
+	runTests(t, tests)
+}
+
+func TestEnumInvalidCases(t *testing.T) {
+	tests := []test{
+		{
+			name: "string mismatch",
```

Comment:
Can you name these tests CamelCase so they fit with the current test names? eg. `StringMismatch`.